pep-0517.txt

```diff
@@ -218,6 +218,23 @@ metadata. The directory passed in by the build frontend MUST be
 identical to the directory created by ``get_wheel_metadata``,
 including any unrecognized files it created.
 
+::
+
+  def prepare_build_files(build_directory, config_settings):
+      ...
+
+Must copy or create any files needed to build this package into
+``build_directory``. For instance, it is expected  that ``pyproject.toml`` will
+exist in the root of this directory after this hook runs. For tools such as
+`setuptools_scm <https://github.com/pypa/setuptools_scm>`_, this may include
+extracting some information from a version control system.
+The ``build_wheel`` hook will subsequently be run from the ``build_directory``
+populated by this hook.
```

Comment:
It's probably obvious, but should be stated explicitly, that ```build_wheel``` run from the directory created by this step must produce an identical wheel to the one produced by running ```build_wheel``` in the original directory.

Should it also be stated that ```pyproject.toml``` must be copied unchanged (to prohibit the sort of silliness I described on distutils-sig)?

But overall, I like this version. It's clean and straightforward.